libcudacxx/include/cuda/std/__cuda/api_wrapper.h

```diff
@@ -47,4 +47,34 @@
     (void) __status;                                   \
   }
 
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+//! @brief `__cuda_set_device_wrapper` is a simple helper that sets the current device to a given target and resets it
+//! back in its destructor
+struct __cuda_set_device_wrapper
```

Comment:
This looks a lot like `cub::SwitchDevice`. If you agree, could you please deprecate/replace/implement in terms of `cub::SwitchDevice`? Or at least create a ticket that we should do this. 

This seems also related to this work: #2040

Reply:
I want this to eventually use the driver API @pciolkosz what do you think about making this publicly available?

Reply:
The driver API interacting version will have a slightly different side effects than this, there is also a good chance it will end up being green context based.
I would say that filing an issue to consider unification in the future is a very good idea. I would not try to be too eager to try to unify this right now while we are still not 100% sure how the final implementation will look like. But once we do, we can sit down and see how we can minimize duplication.
I also have my fingers crossed that this utility will end up being not needed anymore at some point.

Reply:
Also see [`__scoped_device`](https://github.com/NVIDIA/cccl/blob/abf7fbc459ee22223992fb6f57826890d90439b9/cudax/include/cuda/experimental/__device/device.cuh#L58) from #2066

Reply:
I do not want to unify everything now just to then later replace it with whatever we cook up in `cudax`